src/Shared/Model/Providers/NuGetProvider.cs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.Model.Providers;
+
+using Contracts;
+using Metadata;
+using Microsoft.CST.OpenSource.PackageManagers;
+using NLog;
+using NuGet.Packaging.Core;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+using PackageUrl;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Utilities;
+
+public class NuGetProvider : BaseProvider
+{
+    private readonly SourceCacheContext _sourceCacheContext = new();
+    private readonly SourceRepository _sourceRepository = Repository.Factory.GetCoreV3("https://api.nuget.org/v3/index.json");
+    private readonly NuGetLogger _logger = new(LogManager.GetCurrentClassLogger());
+
+    /// <summary>
+    /// Instantiates a new <see cref="NuGetProvider"/>.
+    /// </summary>
+    public NuGetProvider(IHttpClientFactory? httpClientFactory = null)
+    {
+        HttpClientFactory = httpClientFactory ?? new DefaultHttpClientFactory();
+    }
+
+    /// <inheritdoc />
+    public override async Task<string?> DownloadAsync(BaseProjectManager projectManager, PackageURL packageUrl, string targetDirectory, bool doExtract,
+        bool cached = false, CancellationToken cancellationToken = default)
```

Comment:
I don't think this should take a project manager as an argument.  It doesn't appear to actually need the manager.

The only think it used for is `ExtractArchive` which is actually a method implemented on the base class as far as I can tell which just leverages `RecursiveExtractor` and could be refactored into a static helper method.

Reply:
Made it a static method now. Still in BaseProjectManager though. Maybe I should make an ArchiveHelper class? Thoughts?